### ISSUE TITLE
`mergeWith` implementation yields different results from `lodash`'s

### ISSUE THREAD

**Issue by wojtekmaj** [toss/es-toolkit#527]
I'm not yet able to provide a full minimal reproducible example, but I can tell you, `mergeWith` util yields different results than `lodash`'s version in certain cases.

This is the problematic function `mergeWith` was used in:

```ts
import {isEqual}                  from 'es-toolkit/compat';
// TODO: Replace with es-toolkit/compat
import mergeWith                  from 'lodash.mergewith';

export function mergeIntoTarget<T extends object, S extends Array<object>>(target: T, ...sources: S): MergeObjects<S, T> {
  // We need to wrap everything in an object because otherwise lodash fails to merge 2 top-level arrays
  const wrap = <T>(value: T) => ({value});

  const wrappedTarget = wrap(target);
  const wrappedSources = sources.map(source => wrap(source));

  const {value} = mergeWith(wrappedTarget, ...wrappedSources, (targetValue: unknown, sourceValue: unknown) => {
    // We need to preserve comments in custom Array classes such as comment-json's `CommentArray`, so we can't use spread or `Set`s
    if (Array.isArray(targetValue) && Array.isArray(sourceValue)) {
      for (const sourceItem of sourceValue) {
        if (!targetValue.find(targetItem => isEqual(targetItem, sourceItem))) {
          targetValue.push(sourceItem);
        }
      }

      return targetValue;
    }

    return undefined;
  });

  return value;
}
```

and this is the test that started failing:

```ts
import CJSON from 'comment-json';

import * as miscUtils from './miscUtils.ts';

describe(`mergeIntoTarget`, () => {
  it(`should preserve comments when the target is an object created by comment-json`, () => {
    const a = CJSON.parse(`{
      // n
      "n":
      // array
      [
        // 1
        1,
        // 2
        2,
        // 3
        3
      ]
    }`);
    const b = {n: [4, 5, 6]};
    const c = miscUtils.mergeIntoTarget(a, b);

    expect(CJSON.stringify(c, null, 2)).toStrictEqual(CJSON.stringify(CJSON.parse(`{
      // n
      "n":
      // array
      [
        // 1
        1,
        // 2
        2,
        // 3
        3,
        4,
        5,
        6
      ]
    }`), null, 2));
  });
});
```

and `target` is:

```
{
        n: [ 1, 2, 3, 4, 5, 6 ],
        [Symbol(before:n)]: [ [Object] ],
        [Symbol(after-colon:n)]: [ [Object] ]
      },
```

so perhaps it has something to do with Symbols...

**Comment by raon0211** [comment]
This is strange, let me check.

**Comment by raon0211** [comment]
Hi, this bug was resolved in es-toolkit v1.19.0. 

The issue occurred because comments-json adds special symbol properties to the original array, which we hadn’t preserved before. We fixed it by ensuring those properties are now preserved.

**Comment by wojtekmaj** [comment]
I can confirm 1.19.0 fixes my issue! Can't believe you've done this with such a bad issue report :D
